libs/wingc/src/lsp/sync.rs

```diff
@@ -20,7 +20,7 @@ pub struct FileData {
 	/// The diagnostics returned by wingc
 	pub diagnostics: Diagnostics,
 	/// The top scope of the file
-	pub scope: Scope,
+	pub scope: Box<Scope>,
```

Comment:
IIUC this might be a possible use case for `Pin<Box<Scope>>`

Reply:
I think you are right, but I suspect all our `UnsafeRef`s are a use case for `Pin<>` and I'm also not sure it'll compile since I don't think you can pin stuff in a `RefCell`. I'd prefer to just refactor everything to get rid of `UnsafeRef`, but not sure we have time for that now.